src/herramientas/ansoff/ansoff.service.ts

```diff
@@ -107,4 +107,14 @@ export class AnsoffService {
         if (result.deletedCount) return id
         else throw new HttpException('Ansoff not found', HttpStatus.NOT_FOUND)
     }
+
+    async deleteAllWithProjectId(projectId: string) {
+        const result = await this.ansoffModel.deleteMany({ projectId })
+
+        if (result && result.acknowledged) {
+            return projectId
+        } else {
+            throw new HttpException('Ansoff not found', HttpStatus.NOT_FOUND)
```

Comment:
Si no me equivoco esta excepción no sería un not found sino que falló el delete. Si no hay nada que borrar no sería un error, porque puede que no tenga de esa herramienta (ídem con las demás)

Reply:
Si, no me cerro cuando lo hice tmb.. creo que deberia ser un try/catch, y en caso de que el delete falle por falla de conexion, timeout etc, ahi en el catch arrojar un error.

Reply:
O sea tirar una excepción está bien, digo que el error no sería "not found" sino que sería algo como que falló el delete